src/main/java/org/prebid/server/bidder/tappx/TappxBidder.java

```diff
@@ -107,7 +107,12 @@ private String buildEndpointUrl(ExtImpTappx extImpTappx, Integer test) {
             throw new PreBidException("Tappx tappxkey undefined");
         }
 
-        String url = String.format("%s%s/%s?tappxkey=%s", endpointUrl, host, endpoint, tappxkey);
+        String url = endpointUrl + host;
+        if (!host.contains(endpoint)) {
+            url += "/" + endpoint;
+        }
+        url += "?tappxkey=" + tappxkey;
+
```

Comment:
We have many concatenations in this method, can we use StringBuilder?

Reply:
@snahornyi Sure, but that would look somewhere like this:
```
...
StringBuilder urlBuilder = new StringBuilder()
                .append(endpointUrl)
                .append(host);

        if (!host.contains(endpoint)) {
            urlBuilder.append("/").append(endpoint);
        }

        urlBuilder.append("?tappxkey=").append(tappxkey);

        if (test != null && test == 0) {
            int t = (int) System.nanoTime();
            urlBuilder.append("&ts").append(t);
        }

        urlBuilder
                .append("&v=").append(VERSION)
                .append("&type_cnn=").append(TYPE_CNN);

        String url = urlBuilder.toString();
        try {
            HttpUtil.validateUrl(url);
        } catch (IllegalArgumentException e) {
            throw new PreBidException("Not valid url: " + url, e);
        }

        return url;
...
```
Which I think is less cleaner than "old" approach with `url += ...`
I'm trying to say it is not required here - we're not concatenating string in loop. 
Since it's a simple concatenation, Java compiler will do the trick for us. At least for now I don't see any problem with this approach. 
If not - let me know :)